mne/connectivity/cfc.py

```diff
@@ -0,0 +1,529 @@
+
+    return times, ampmat, traces, avg, trigger_inds
+
+
+def compute_cfc_stats(sfreq, ampmat, traces, freqs, trigger_inds,
+                      n_surrogates=10000, alpha=0.001, random_state=None):
+    """ Stats function for cross frequency coupling.
+
+    Parameters
+    ----------
+    sfreq : float
+        Sampling frequency.
+    ampmat : ndarray, shape (nfreq, n_samples)
+        Amplitude traces.
+    traces : ndarray, shape (n_epochs, n_times)
+        Normalized amplitude traces.
+    freqs : array, shape (n_freqs,)
+        Frequency points across range at which amplitudes are computed.
+    trigger_inds : array (trigger_inds,)
+        Indices of maxima in the phase signal.
+    n_surrogates : int
+        Number of surrogates. Default 10 ** 4.
+    alpha : float
+        Error rate allowed. Default 0.001.
+    random_state : None | int | instance of np.random.RandomState
+        np.random.RandomState to initialize the noise estimation.
+        Default value is None.
+
+    Returns
+    -------
+    ztraces : ndarray, shape (n_freqs, n_times)
+        Z-scored amplitude traces.
+    z_threshold : float
+        FDR corrected threshold for z-scored amplitude traces.
+    """
+
+    n_freqs = len(freqs)
+    n_samples = ampmat.size / n_freqs
+    epoch_length = traces.size / n_freqs
+
+    # permutation resampling for statistical significance:
+    # only need to compute mean and variance of surrogate distributions at one
+    # time point for all frequencies, since the surrogate trigger events could
+    # have occurred at any point
+    rng = check_random_state(random_state)
+    shifts = np.floor(rng.rand(2 * n_surrogates) * n_samples)
+    # get rid of trigger shifts within one epoch of actual occurence
+    shifts = shifts[shifts > epoch_length]
+    shifts = shifts[shifts < n_samples - epoch_length]
+
+    sur_distributions = np.zeros((n_freqs, n_surrogates), dtype=np.float)
+    for sur in range(n_surrogates):
+        if n_surrogates - sur + 1 % 100 == 0:
+            logger.info(n_surrogates - sur + 1)
+        # circular shift
+        sur_triggers = np.mod(trigger_inds + shifts[sur], n_samples)
+        sur_triggers = sur_triggers.astype(np.int)
+        sur_distributions[:, sur] = np.mean(ampmat[:, sur_triggers], axis=1)
+
+    sur_fits_mean = np.mean(sur_distributions, axis=1)
+    sur_fits_std = np.std(sur_distributions, axis=1)
+
+    ztraces = (traces - sur_fits_mean[:, None]) / sur_fits_std[:, None]
+
+    # Compute FDR threshold
+    p_values = stats.norm.pdf(ztraces)
+    accept, _ = fdr_correction(p_values, alpha=alpha)
+    # check if any acceptable values are found
+    if not accept.any():
+        raise RuntimeError('No significant cross frequency coupling found'
+                           ' try changing the frequency range.')
+    z_threshold = np.abs(ztraces[accept]).min()
+
+    return ztraces, z_threshold
+
+
+def phase_amplitude_coupling(data, sfreq, l_phase_freq, h_phase_freq,
```

Comment:
I think I am starting to understand the differences here. `phase_amplitude_coupling` is basically for calculating the amplitude distribution as a function of phase, while `cross_frequency_coupling` is for calculating the evoked spectral energy time-locked to phase peaks, and is only for visualization. Is this correct?